kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/importer/sink/PipelineDataSourceSink.java

```diff
@@ -133,67 +136,47 @@ private void tryFlush(final DataSource dataSource, final List<DataRecord> buffer
         }
     }
     
-    private void doFlush(final DataSource dataSource, final List<DataRecord> buffer) throws SQLException {
+    private void doFlush(final DataSource dataSource, final List<DataRecord> buffer, final boolean enableTransaction) throws SQLException {
         try (Connection connection = dataSource.getConnection()) {
-            connection.setAutoCommit(false);
-            switch (buffer.get(0).getType()) {
-                case IngestDataChangeType.INSERT:
-                    if (null != rateLimitAlgorithm) {
-                        rateLimitAlgorithm.intercept(JobOperationType.INSERT, 1);
-                    }
-                    executeBatchInsert(connection, buffer);
-                    break;
-                case IngestDataChangeType.UPDATE:
-                    if (null != rateLimitAlgorithm) {
-                        rateLimitAlgorithm.intercept(JobOperationType.UPDATE, 1);
-                    }
-                    executeUpdate(connection, buffer);
-                    break;
-                case IngestDataChangeType.DELETE:
-                    if (null != rateLimitAlgorithm) {
-                        rateLimitAlgorithm.intercept(JobOperationType.DELETE, 1);
-                    }
-                    executeBatchDelete(connection, buffer);
-                    break;
-                default:
-                    break;
+            if (enableTransaction) {
+                connection.setAutoCommit(false);
             }
-            connection.commit();
-        }
-    }
```

Comment:
Could we just remove added parameter `enableTransaction`? Judge whether to enable transaction by `buffer.size`. And keep these code not deleted. It could be simpler.